tests/func-tests/node_placement_test.go

```diff
@@ -17,39 +22,146 @@ import (
-	"kubevirt.io/client-go/kubecli"
-	"kubevirt.io/kubevirt/tests/flags"
-
 	tests "github.com/kubevirt/hyperconverged-cluster-operator/tests/func-tests"
+	"kubevirt.io/client-go/kubecli"
 )
 
-const nodePlacementSamplingSize = 20
-
-var _ = Describe("[rfe_id:4356][crit:medium][vendor:cnv-qe@redhat.com][level:system]Node Placement", func() {
+const (
+	nodePlacementSamplingSize = 20
+	hcoLabel                  = "node.kubernetes.io/hco-test-node-type"
+	infra                     = "infra"
+	workloads                 = "workloads"
+	group                     = "hco.kubevirt.io"
+	version                   = "v1beta1"
+	kind                      = "HyperConverged"
+	hcoResource               = "hyperconvergeds"
+	name                      = "kubevirt-hyperconverged"
+)
 
+var _ = Describe("[rfe_id:4356][crit:medium][vendor:cnv-qe@redhat.com][level:system]Node Placement", Ordered, func() {
 	var workloadsNode *v1.Node
 
 	tests.FlagParse()
 	client, err := kubecli.GetKubevirtClient()
 	kvtutil.PanicOnError(err)
 
-	workloadsNodes, err := client.CoreV1().Nodes().List(context.TODO(), k8smetav1.ListOptions{
-		LabelSelector: "node.kubernetes.io/hco-test-node-type==workloads",
-	})
-	kvtutil.PanicOnError(err)
-
-	if workloadsNodes != nil && len(workloadsNodes.Items) == 1 {
-		workloadsNode = &workloadsNodes.Items[0]
+	// store the existing HCO CR obtained in BeforeALl and revert in AfterAll stage
+	originalHco := &hcov1beta1.HyperConverged{}
+
+	BeforeAll(func() {
+		nodes, err := client.CoreV1().Nodes().List(context.TODO(), k8smetav1.ListOptions{LabelSelector: "node-role.kubernetes.io/worker"})
+		kvtutil.PanicOnError(err)
+		totalNodes := len(nodes.Items)
+
+		// Label all but first node with "node.kubernetes.io/hco-test-node-type=infra"
+		// We are doing this to remove dependency of this Describe block on a shell script that
+		// labels the nodes this way
+		for i := 0; i < totalNodes; i++ {
+			err = setHcoNodeTypeLabel(client, &nodes.Items[i], infra)
+			kvtutil.PanicOnError(err)
+			if i == totalNodes-1 {
+				break
+			}
+		}
+		// Label the last node with "node.kubernetes.io/hco-test-node-type=workloads"
+		err = setHcoNodeTypeLabel(client, &nodes.Items[totalNodes-1], workloads)
+		kvtutil.PanicOnError(err)
+
+		Expect(client.RestClient().
+			Get().
+			Resource(hcoResource).
+			Name(name).
+			Namespace(flags.KubeVirtInstallNamespace).
+			AbsPath("/apis", hcov1beta1.SchemeGroupVersion.Group, hcov1beta1.SchemeGroupVersion.Version).
+			Timeout(10 * time.Second).
+			Do(context.TODO()).
+			Into(originalHco),
+		).To(Succeed())
+		// modify the "infra" and "workloads" keys
+		infraVal := hcov1beta1.HyperConvergedConfig{NodePlacement: &sdkapi.NodePlacement{}}
+		workloadsVal := hcov1beta1.HyperConvergedConfig{NodePlacement: &sdkapi.NodePlacement{}}
+
+		infraVal.NodePlacement.NodeSelector = map[string]string{hcoLabel: infra}
+		workloadsVal.NodePlacement.NodeSelector = map[string]string{hcoLabel: workloads}
```

Comment:
Let's not split it. You can try something like this:
```suggestion
		infraVal := hcov1beta1.HyperConvergedConfig{
			NodePlacement: &sdkapi.NodePlacement{
				NodeSelector: map[string]string{hcoLabel: infra},
			},
		}
		
		workloadsVal := hcov1beta1.HyperConvergedConfig{
			NodePlacement: &sdkapi.NodePlacement{
				NodeSelector: map[string]string{hcoLabel: workloads},
			},
		}
```